security/iap-demo/main.tf

```diff
@@ -0,0 +1,240 @@
+locals {
+  # Set these parameters if establishing Cloud VPN HA/dynamic routing
+  asn_local           = var.asn_local
+  asn_peer            = var.asn_peer
+  billing_account_id  = var.billing_account_id
+  tunnel_0_link_range = var.tunnel_0_link_range
+  tunnel_1_link_range = var.tunnel_1_link_range
+  # Set these parameters if establishing Cloud VPN Classic/static routing
+  # General parameters
+  cert_domains             = var.cert_domains
+  cert_name                = var.cert_name
+  dns_zone                 = var.dns_zone
+  dns_resolvers            = var.dns_resolvers
+  mappings                 = var.mappings
+  master_authorized_ranges = var.master_authorized_ranges
+  name                     = var.name
+  peer_vpn_ip              = var.peer_vpn_ip
+  policy_name              = var.policy_name
+  project_id               = var.project_id
+  project_owners           = var.project_owners
+  range_master             = var.range_master
+  range_nodes              = var.range_nodes
+  range_pods               = var.range_pods
+  range_services           = var.range_services
+  region                   = var.region
+  root_node                = var.root_node
+  shared_secret            = var.shared_secret
+  support_email            = var.support_email
+  web_user_principals      = var.web_user_principals
+
+  project_services = [
+    "cloudresourcemanager.googleapis.com",
+    "compute.googleapis.com",
+    "container.googleapis.com",
+    "dns.googleapis.com",
+    "iam.googleapis.com",
+    "iap.googleapis.com",
+    "logging.googleapis.com",
+    "monitoring.googleapis.com",
+  ]
+}
+
+
+# Project where resources will be created
+module "project-iap" {
```

Comment:
Can we add a `project_create` variable passed in to the module, to allow using a pre-existing project? Services will then need to be configured to not disable on destroy, and non disable dependent services.